include/micm/solver/solver.hpp

```diff
@@ -15,26 +17,30 @@ namespace micm
     std::size_t number_of_species_;
     std::size_t number_of_reactions_;
     StateParameters state_parameters_;
-    SolverPolicy solver_;
+    std::vector<micm::Process> processes_;
 
    public:
+    SolverPolicy solver_;
+
     Solver(
-        SolverPolicy solver,
+        SolverPolicy&& solver,
         StateParameters state_parameters,
         std::size_t number_of_grid_cells,
         std::size_t number_of_species,
-        std::size_t number_of_reactions)
-        : solver_(solver),
+        std::size_t number_of_reactions,
+        std::vector<micm::Process> processes)
+        : solver_(std::move(solver)),
           number_of_grid_cells_(number_of_grid_cells),
           number_of_species_(number_of_species),
           number_of_reactions_(number_of_reactions),
-          state_parameters_(state_parameters)
+          state_parameters_(state_parameters),
+          processes_(std::move(processes))
     {
     }
 
-    void Solve(double time_step, StatePolicy& state)
+    SolverResult Solve(double time_step, StatePolicy& state)
```

Comment:
Should we call `CalculateRateConstants` here so that it doesn't need to be called by the user? I'm not sure what the exact requirements for the gpu solver will be, but maybe they don't require a precondition to the solve function?

Reply:
I think the reasoning for having the user do this is that for the CUDA solver, the rate constants have to be calculated on the host, and then synced to the device. Doing this in the `Solver` would require the `Solver` to do something special for a CUDA ODE solver. My vote would be to leave this in the user space and make an issue for calculating rate constants on the device. At that point we could move in into `Solver::Solve()` with no custom logic for CUDA solvers. What do you think?

Reply:
That's fine with me